### PR TITLE
Update deps and scala versions

### PR DESCRIPTION
**Pull request by relrod**
- Include builds for 2.11 and 2.10.4
- Bump scalaz version

### CODE REVIEW

File: project/build.scala [hunk 1]
```diff
@@ -4,17 +4,17 @@ import Keys._
...
 
   def specsVersion(scalaVersion: String) =
     if (scalaVersion startsWith "2.9") "1.12.4.1" else "1.12.3"
 
   lazy val standardSettings = Defaults.defaultSettings ++ List[Sett](
     organization := "org.purefn"
-  , scalaVersion := "2.10.1"
-  , crossScalaVersions := List("2.9.2", "2.9.3", "2.10.1")
+  , scalaVersion := "2.10.4"
+  , crossScalaVersions := List("2.9.2", "2.9.3", "2.10.1", "2.10.4", "2.11.0")
```
Comment (larsrh) [inline project/build.scala]: Cross-building against multiple 2.10.x versions is unneeded. A simple `2.10.4` should suffice. (Same goes for the Travis configuration.)

**Comment by relrod** [comment]
@larsrh fixed - no idea how to get the testsuite to run on both 2.11 and 2.10 though. I tried updating specs2 and scalacheck and it fails on 2.10 and 2.9 under various version combinations, but the current combination fails on 2.11 because there is no scalacheck 1.10.x for Scala 2.11.

**Comment by markhibberd** [comment]
@CodeBlock You will probably need to switch the versions depending on the scala version, there are various levels of sophistication in builds around the place, but the argonaut build has a working 2.9/2.10/2.11 versions:
- https://github.com/argonaut-io/argonaut/blob/series/6.0.x/project/build.scala#L36-L39 (the scalaz dep pulls in the correct scalacheck)
- https://github.com/argonaut-io/argonaut/blob/series/6.0.x/project/Tools.scala

Getting the right versions is a bit like walking a tight rope.

**Comment by relrod** [comment]
@markhibberd awesome, I will give this a shot later today. Thanks!

**Comment by relrod** [comment]
Well that got one step closer, but still failing on 2.11 for me. Might have to do with `bytestring-scalacheck-binding` depending on a specific version of scalacheck?

**Pull request closed without merging**